synapsebridgehelpers/export_tables.py

```diff
@@ -0,0 +1,342 @@
+    if source_cols is None:
+        source_cols = syn.getTableColumns(source)
+    if target_cols is None:
+        target_cols = syn.getTableColumns(target)
+    for action, cols in schema_comparison.items():
+        if action == "added":
+            added_columns = list(filter(lambda c : c["name"] in cols,
+                                      source_cols))
+            target_schema.addColumns(added_columns)
+        if action == "removed":
+            removed_columns = filter(lambda c : c["name"] in cols, target_cols)
+            for c in removed_columns:
+                target_schema.removeColumn(c)
+        if action == "modified":
+            modified_source_columns = list(filter(lambda c : c["name"] in cols,
+                                                  source_cols))
+            modified_target_columns = list(filter(lambda c : c["name"] in cols,
+                                                  target_cols))
+            for c in modified_target_columns:
+                target_schema.removeColumn(c)
+            target_schema.addColumns(modified_source_columns)
+        if action == "renamed":
+            for target_name, source_name in cols.items():
+                renamed_source_column = next(
+                        filter(lambda c : c["name"] == source_name, source_cols))
+                renamed_target_column = next(
+                        filter(lambda c : c["name"] == target_name, target_cols))
+                target_schema.removeColumn(renamed_target_column)
+                target_schema.addColumn(renamed_source_column)
+    target_schema = syn.store(target_schema)
+    return target_schema
+
+
+def export_tables(syn, table_mapping, target_project=None, update=True,
+                  reference_col="recordId", copy_file_handles=True, **kwargs):
+    """Export records from one Synapse table to another. Or transfer tables
+    to completely new tables in a separate project.
+
+    Parameters
+    ----------
+    syn : synapseclient.Synapse
+    table_mapping : dict, list, or str
+        If exporting records of one or more tables to other, preexisting tables,
+        table_mapping is a dictionary containing Synapse ID key/value mappings
+        from source to target tables. If exporting table records to not yet
+        created tables in a seperate project, table_mapping can be a list or
+        string.
+    target_project : str
+        If exporting table records to not yet created tables in a seperate
+        project, specify the target projects Synapse ID here.
+    update : bool, defaults to True
+        When exporting records of one or more tables to other, preexisting
+        tables, whether to append new records to these tables or completely
+        overwrite the table records.
+    referenceCol : str
+        If `update` is True, use this column as the table index to determine
+        which records are already present in the target table.
+    copy_file_handles : bool, defaults to True
```

Comment:
Is this limited to single column or multiple.  There are some projects with repeated recordIds with different data - I suspect these were due to an early bridge bug but it is worth noting that recordId is not always unique.

Reply:
Nope, we can pass a list here if necessary. I'll update the docstring.